collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -590,22 +590,22 @@ private void loadFromIntent(Intent intent) {
             formPath = intent.getStringExtra(EXTRA_TESTING_PATH);
 
         } else if (uriMimeType != null && uriMimeType.equals(InstanceColumns.CONTENT_ITEM_TYPE)) {
-            FormInfo formInfo = ContentResolverHelper.getFormDetails(uri);
+            Instance instance = new DatabaseInstancesRepository().get(ContentUriHelper.getIdFromUri(uri));
 
-            if (formInfo == null) {
+            if (instance == null) {
                 createErrorDialog(getString(R.string.bad_uri, uri), true);
                 return;
             }
 
-            instancePath = formInfo.getInstancePath();
-            List<Form> candidateForms = formsRepository.getAllByFormIdAndVersion(formInfo.getFormId(), formInfo.getFormVersion());
+            instancePath = instance.getInstanceFilePath();
+            List<Form> candidateForms = formsRepository.getAllByFormIdAndVersion(instance.getJrFormId(), instance.getJrVersion());
 
             if (candidateForms.isEmpty()) {
                 createErrorDialog(getString(
                         R.string.parent_form_not_present,
-                        formInfo.getFormId())
-                                + ((formInfo.getFormVersion() == null) ? ""
-                                : "\n" + getString(R.string.version) + " " + formInfo.getFormVersion()),
+                        instance.getJrFormId())
```

Comment:
That's a good question. I'm guessing it stands for "Java Rosa"? Really it feels like it should be "Open Rosa" right? Maybe (for `Form` and `Instance`) we should swap where the specifier is and have `id` (String), `version` (String) and `dbId` (Long)? I'm interested to tag @lognaturel in here as she might have some context around the current naming.

Reply:
I like using `formId` because that matches how it's referred to in documentation and other parts of the system. I have made that change when I've touched related code. I do prefer `dbId` to just `id` as well.

Reply:
Great. I'll make that change while I'm here.

Reply:
We still have columns in forms databse like:
const val JR_FORM_ID = "jrFormId"
const val JR_VERSION = "jrVersion"

should we rename them too? Probably not because external apps may relay on the existing naming?

Reply:
Good point. Yeah I was assuming we'd just leave those in place for now. Our own app shouldn't really have to care about the column names outside the `DatabaseFormsRepository` anyway.